src/middleware.ts

```diff
@@ -0,0 +1,12 @@
+import { NextRequest, NextResponse } from 'next/server';
+
+export async function middleware(request: NextRequest) {
+  const currentUser = request.cookies.get('auth-user')?.value;
+
+  if (!currentUser && request.nextUrl.pathname.startsWith('/')) {
+    return NextResponse.redirect(new URL('/auth', request.nextUrl.origin));
+  }
+}
+export const config = {
+  matcher: ['/categories', '/category', '/list', '/about', '/'],
```

Comment:
Here In added routes that need to be protected if a user is not logged in. Any other I missed?

Reply:
'/contact'

but we could leave it open along with 'about'. As you wish.